psi4/src/psi4/libfock/solver.cc

```diff
@@ -253,9 +253,9 @@ void CGRSolver::guess() {
 }
 void CGRSolver::residual() {
     for (size_t N = 0; N < b_.size(); ++N) {
-        r_[N]->copy(Ap_[N].get());
+        r_[N]->copy(*Ap_[N]);
```

Comment:
There's a lot of removing .get()s and adding dereferencing *, which isn't the direction of code changes I'm used to seeing. Looks like a switch from the stored (not managed) pointer to the value pointed at by the shared_ptr, without changing the fn signature. Is there a term for this procedure I can look up or a quick explanation?

Reply:
The quick explanation is that the function signature actually did change.

Old code had one version of `copy` where the argument was passed by _pointer_ and one where it was passed by _reference_. `copy` doesn't need to manage memory or accept a `null` value, so the new code exclusively has the pass by _reference_.

Old code used pass-by-pointer (even though there was a perfectly good pass-by-reference), new code uses pass-by-reference.